pkg/driver/driver_aws.go

```diff
@@ -289,14 +289,20 @@ func (d *AWSDriver) Delete(machineID string) error {
 		if err != nil {
 			if aerr, ok := err.(awserr.Error); ok {
 				switch aerr.Code() {
+				case "InvalidInstanceAttributeValue":
```

Comment:
Can we please dig a little deep before fixing this issue. I am not really sure if we should always skip if `InvalidInstanceAttributeValue ` is seen. 
- I am curious, why would the machine-object be deleted if the deletion of VM failed here. 
- And do we really need this code-path any more, as this was added only for the temporary use-case?

Reply:
Control goes into this part of the code - https://github.com/gardener/machine-controller-manager/blob/master/pkg/controller/machine.go#L434-L451. So the machine object would've been deleted by the machineSet controller as it wouldn't have observed a `providerID`.

Reply:
But returning error here doesn't help as the machine object is already deleted.

Reply:
> And do we really need this code-path any more, as this was added only for the temporary use-case?
I think we might still need it as there may be clusters where the machines were created using the faulty APIs. Those machines might still be running for a cluster.

Reply:
Thanks. Sounds good, we might then want to fix the control-loop of machine later. I'd resolve this conversation for now.